### PR TITLE
Release 3.1.2

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 

- Add changelog for v3.1.2
- Change the version to v3.1.2
- Bump spark version from 3.2.2 to 3.2.3, 3.3.0 to 3.3.1

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### CODE REVIEW

**Review by Daemonxiao — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
 
+## [TiSpark 3.1.2] 2022-12-12
+### New Features
+- Support write into partition table with new collations [#2570](https://github.com/pingcap/tispark/pull/2570)
+- Support read TiFlash load balancing with Round-Robin strategy [#2576](https://github.com/pingcap/tispark/pull/2576)
+- Support partition pruning with to_days function when read from TiKV/TiFlash [#2594](https://github.com/pingcap/tispark/pull/2594) [#2600](https://github.com/pingcap/tispark/pull/2594)
+
+### Fixes
+- Fix CBO to let TiSpark choose the min cost between TiKV table scan, TiKV index scan and TiFlash scan correctly [#2568](https://github.com/pingcap/tispark/pull/2568)
+- FIx the `region not find` error when reading from TiKV [#2575](https://github.com/pingcap/tispark/pull/2575)
```
Comment (shiyuhang0) [inline CHANGELOG.md:12]:
```suggestion
- Fix the `region not find` error when reading from TiKV [#2575](https://github.com/pingcap/tispark/pull/2575)
```

**Comment by shiyuhang0** [comment]
/run-all-tests profile=spark-3.2 tidb=release-6.1 tikv=release-6.1

**Comment by shiyuhang0** [comment]
/run-all-tests profile=spark-3.3 tidb=release-4.0 tikv=release-4.0

**Review by zhangyangyu — APPROVED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Review by zhangyangyu — APPROVED** [review]

**Pull request merged**